crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -709,20 +727,42 @@ impl Space {
         ep_hnd: Tx2EpHnd<wire::Wire>,
         config: Arc<KitsuneP2pConfig>,
     ) -> Self {
-        let gossip_mod_fact = if &config.tuning_params.gossip_strategy == "simple-bloom" {
-            crate::gossip::simple_bloom::factory()
-        } else {
-            panic!(
-                "unknown gossip strategy: {}",
-                config.tuning_params.gossip_strategy
-            );
-        };
-        let gossip_mod = gossip_mod_fact.spawn_gossip_task(
-            config.tuning_params.clone(),
-            space.clone(),
-            ep_hnd.clone(),
-            evt_sender.clone(),
-        );
+        let gossip_mod = config
+            .tuning_params
+            .gossip_strategy
+            .split(',')
+            .flat_map(|module| match module {
+                "simple-bloom" => vec![(
+                    GossipModuleType::Simple,
+                    crate::gossip::simple_bloom::factory(),
+                )],
+                "sharded-gossip" => vec![
+                    (
+                        GossipModuleType::ShardedRecent,
+                        crate::gossip::sharded_gossip::recent_factory(),
+                    ),
+                    // (
+                    //     GossipModuleType::ShardedHistorical,
+                    //     crate::gossip::sharded_gossip::historical_factory(),
+                    // ),
+                ],
+                _ => {
+                    tracing::warn!("unknown gossip strategy: {}", module);
```

Comment:
shouldn't this be fatal? I'd rather add a dummy gossip module if we need a kitsune that doesn't gossip for testing.